File: problems/Missing Element in Sorted Array/missingElement.cs
public class Solution {
    public int MissingElement(int[] nums, int k) {
        int n = nums.Length;
        int left = 0;
        int right = n - 1;

        if (k > missing(n - 1, nums)) {
            return nums[n - 1] + k - missing(n - 1, nums);
        }

        while (left != right) {
            int mid = left + ((right - left) >> 1);

            if (k > missing(mid, nums)) {
                left = 1 + mid;
            } else {
                right = mid;
            }
        }

        return nums[left - 1] + k - missing(left - 1, nums);
    }

    private int missing(int idx, int[] nums) {
        return nums[idx] - nums[0] - idx;
    }
}

File: problems/Missing Element in Sorted Array/missingElement.java
class Solution {
    public int missingElement(int[] nums, int k) {
        int n = nums.length;

        if (k > missing(nums, n - 1)) {
            return nums[n - 1] + k - missing(nums, n - 1);
        }

        int left = 0;
        int right = n;

        while (left + 1 < right) {
            int mid = left + (right - left) / 2;

            if (k > missing(nums, mid)) {
                left = mid;
            } else {
                right = mid;
            }
        }

        return nums[left] + k - missing(nums, left);
    }

    int missing(int[] nums, int idx) {
        return nums[idx] - nums[0] - idx;
    }
}
